tests/integration/synapseclient/models/async/test_dataset_async.py

```diff
@@ -549,3 +549,424 @@ async def test_rename_column(self, project_model: Project) -> None:
         )
         assert new_dataset_instance.columns[new_column_name] is not None
         assert old_column_name not in new_dataset_instance.columns
+
+
+class TestDatasetCollection:
```

Comment:
There should be a few tests to verify snapshot functionality is working as expected:


1. A dataset collection with a few dataset is snapshot. The old version should remain the same even after the dataset collection has modifications, and querying the new version shows the newest changes
2. Verify that you can snapshot a dataset and add that specific version to the dataset collection
3. Verify that you can have multiple versions of the same dataset added to the dataset collection

Reply:
There are limits on this imposed by the Synapse backend. You cannot have more than one version of the same file in a dataset, and you cannot have more than one version of a dataset in a dataset collection
```
synapseclient.core.exceptions.SynapseHTTPError: 400 Client Error: Each dataset item must have a unique entity ID.  Duplicate: syn64893554
```

```
synapseclient.core.exceptions.SynapseHTTPError: 400 Client Error: Each dataset collection item must have a unique entity ID.  Duplicate: syn66271126
```

I will make sure that snapshot functionality is adequately tested otherwise though

Reply:
Good to know. I figured that it would be possible, but thanks for verifying this